plugins/database/postgresql/postgresql.go

```diff
@@ -44,7 +44,7 @@ func new() *PostgreSQL {
 	connProducer.Type = postgreSQLTypeName
 
 	credsProducer := &credsutil.SQLCredentialsProducer{
-		DisplayNameLen: 8,
+		DisplayNameLen: 30,
```

Comment:
Hi @simonmacklin ! Thanks for this PR!

This is interesting. It seems like a very small change to change the length from 8 to 30, but I actually do wonder if it could have a negative impact on someone who has created functionality around expecting the generated name to be a certain length. Also, I'm not certain about all the acceptable lengths over all versions of Postgres.

What do you think about making it so that it defaults to 8, but if the user provides a certain config parameter, it can be overridden?

The database secrets engine mainly only ingests connection information, but I'm wondering if we could do it by changing something like this:
```
$ vault write database/config/my-database \
    plugin_name="..." \
    connection_url="..." \
    allowed_roles="..." \
    username="..." \
    password="..."
```
To support an additional parameter like `display_name_len`. We could grab it in the `Initialize` method noted [here](https://www.vaultproject.io/docs/secrets/databases/custom/#plugin-interface) (though it's referred to as `Init` there, which is outdated). Then we could store it in memory and use it for generating the display name.

That does have a problem, though, where in a replicated environment, some Vault instances server _reads and writes_, while others only serve _reads_. So, I'm actually not certain if that approach would work in an replicated environment. :thinking: 

Now that I better understand what we're wanting to do here, I need to think on the best way to do it. It may actually be a bigger lift than originally thought.

Reply:
Hi @tyrannosaurus-becks 

Thanks again for your reply :) 

Regards to the total username length.  How I read the code is the length will always be 63 chars long.  Never smaller or larger.  Increasing the displayname var just means it will take up 30 of the total 63 char length.

Making this configurable would be super and I would be more then happy to do the work if that helps.  

Simon

Reply:
Ah, gotcha. That could be okay then. Would you be willing to add some test coverage of this config showing what the usernames would now look like with this configuration, and various inputs to the `GenerateUsername` method?